### ISSUE TITLE
External function is not supported for resource methods

### ISSUE THREAD

**Issue by TharmiganK** [ballerina-platform/ballerina-lang#37020]
**Description:**

With the [`Allow client objects to have resource methods #36024`](https://github.com/ballerina-platform/ballerina-lang/issues/36024) improvement and the client resource access action support given in this [PR](https://github.com/ballerina-platform/ballerina-lang/pull/36959), there is a requirement to add this support to HTTP client through the following resource signature : 

```ballerina
 public type SimpleParamType boolean|int|float|decimal|string;
 public type ParamType SimpleParamType[]|SimpleParamType;
 public type TargetType typedesc<http:Response|anydata>;

 public type Params record {|
     never headers?;
     never targetType?;
     ParamType...;
 |};

 // Should be extern since the return type is dependent
 public resource function get [string ...path](map<string|string[]>? headers = (),
      TargetType targetType = <>, *Params params) returns targetType|error = external;
```

**Describe your problem(s)**

But having the resource function as external is giving the following error :
```
external resource functions are not supported by the implementation
```

**Comment by rdulmina** [comment]
Shouldn't it be the following signature as proposed in the doc
```bal

 public resource function get [string ...path](map<string|string[]>? headers = (),
                                               TargetType targetType = <>, *Params params) returns targetType|error {
    return self.get(makeRequestPath(path, params), headers, targetType);
 }

```
`self.get` will have a `external-function-body`

**Comment by rdulmina** [comment]
Decided to allow `external-function-body` for resource methods inside client objects

**Comment by daneshk** [comment]
@rdulmina is it only for the resource methods? or both resource and remote methods inside client objects

**Comment by daneshk** [comment]
Better if we can get this support for both resource and remote methods inside client objects. We have a use-case in persist module that needs this support in remote methods.

**Comment by rdulmina** [comment]
We already allow external function body for remote methods

**Comment by daneshk** [comment]
@rdulmina, not external function bodies, are we support writing dependently type functions in Ballerina?

**Comment by rdulmina** [comment]
Fixed with  #37042
